src/main/java/com/code_intelligence/jazzer/runtime/ClojureLangHooks.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.runtime;
+
+import com.code_intelligence.jazzer.api.HookType;
+import com.code_intelligence.jazzer.api.MethodHook;
+import java.lang.invoke.MethodHandle;
+import java.util.*;
+
+@SuppressWarnings("unused")
+final public class ClojureLangHooks {
+    static ThreadLocal<WeakHashMap<Object, String>> markedObjects = ThreadLocal.withInitial(() -> new WeakHashMap<Object, String>());
```

Comment:
Note that using a `WeakHashMap` is only correct if the key type does not override `equals`. Could you add a comment pointing to the definition of the type we expect to be stored in this map?

Reply:
Yes, will do!

Reply:
Done!